examples/path_requests_run.py

```diff
@@ -59,15 +61,52 @@ def requests_from_json(json_data,equipment):
         params['loose_list'] = [n['unnumbered-hop']['hop-type'] for n in nd_list]
         params['spacing'] = req['path-constraints']['te-bandwidth']['spacing']
 
+        # recover trx physical param (baudrate, ...) from type and mode
+        # in trx_mode_params optical power is read from equipment['SI']['default'] and
+        # nb_channel is computed based on min max frequency and spacing
         trx_params = trx_mode_params(equipment,params['trx_type'],params['trx_mode'],True)
         params.update(trx_params)
-        params['power'] = req['path-constraints']['te-bandwidth']['output-power']
-        params['nb_channel'] = req['path-constraints']['te-bandwidth']['max-nb-of-channel']
+        # optical power might be set differently in the request. if it is indicated then the 
+        # params['power'] is updated
+        if req['path-constraints']['te-bandwidth']['output-power']:
+            params['power'] = req['path-constraints']['te-bandwidth']['output-power']
+        # same process for nb-channel
+        if req['path-constraints']['te-bandwidth']['max-nb-of-channel'] :
+            # check if requested nb_channels is consistant with baudrate and min-max frequencies
+            min_recommanded_spacing = automatic_spacing(trx_params['baud_rate'])
+            fmin = trx_params['frequency']['min']
+            fmax = trx_params['frequency']['max']
+            max_recommanded_nb_channels = automatic_nch(fmin,fmax,
+                min_recommanded_spacing)
+                
+            if req['path-constraints']['te-bandwidth']['max-nb-of-channel'] <= max_recommanded_nb_channels :
+                params['nb_channel'] = req['path-constraints']['te-bandwidth']['max-nb-of-channel']
+            else:
+                temp = params['baud_rate']
 
+                msg = f'Requested channel number is not consistant with frequency range  : \
```

Comment:
Something like this might be easier to read (using `textwrap.dedent`)…

```suggestion
                msg = dedent(f'''
                Requested channel number is not consistent with frequency range:
                {fmin*1e-12} THz, {fmax*1e-12} THz and baud rate: {temp*1e-9} GHz
                min recommanded spacing is {min_recommanded_spacing}
                max recommanded nb of channels is {max_recommanded_nb_channels}
                Computation stopped.''')
```